flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/source/assigner/SplitAssigner.java

```diff
@@ -115,4 +115,7 @@ default void onCompletedSplits(Collection<String> completedSplitIds) {}
    * snapshots and splits, which defeats the purpose of throttling.
    */
   int pendingSplitCount();
+
+  /** Pending records count */
```

Comment:
How good is the `split.task().estimatedRowsCount())` which we are using for the calculations? Shall we state in the comments, or in the method name, that the value is estimated? 🤔

Reply:
I won't worry too much. the `estimated` part comes from a split large file. there is no record count for each split/chunk. hence the record count is estimated based on ratio of the split bytes / file bytes.

Reply:
I still think that the method name should contain `estimated`.

Reply:
I think of it as more an implementation detail whether it is estimated or not--perhaps there could be a more accurate way of computing this value in the future. Rather than leave the comment in the interface, are you ok if we just leave the comment on the implementing method?

Reply:
This is a public facing interface.
What about:
```
/** Pending records count. Could be an estimation if exact numbersare not available*/
```

I am not comfortable stating one thing in the comments/docs and doing other things in the implementation.